From: "Roman Pišl" <rpisl@seznam.cz>
Subject: [PATCH 2/2] ntdll: Add DECLSPEC_HOTPATCH to strrchr.
Message-Id: <20211208225259.68309-2-rpisl@seznam.cz>
Date: Wed,  8 Dec 2021 23:52:59 +0100
In-Reply-To: <20211208225259.68309-1-rpisl@seznam.cz>
References: <20211208225259.68309-1-rpisl@seznam.cz>

I see no other way how to make Clang ASAN work.
ASAN recognizes first instruction(s) and inserts trampoline, but the loop
in the function body jumps back to the start of the function (and hangs)
or somewhere inside the trampoline (that leads to invalid instruction).

Other fixes can be performed on the ASAN side, see the bug report.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=50993
Signed-off-by: Roman Pišl <rpisl@seznam.cz>
---
 dlls/ntdll/string.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ntdll/string.c b/dlls/ntdll/string.c
index 11111111111..11111111111 100644
--- a/dlls/ntdll/string.c
+++ b/dlls/ntdll/string.c
@@ -437,7 +437,7 @@ char * __cdecl strpbrk( const char *str, const char *accept )
 /*********************************************************************
  *                  strrchr   (NTDLL.@)
  */
-char * __cdecl strrchr( const char *str, int c )
+char * __cdecl DECLSPEC_HOTPATCH strrchr( const char *str, int c )
 {
     char *ret = NULL;
     do { if (*str == (char)c) ret = (char *)(ULONG_PTR)str; } while (*str++);

-- 
2.36.0

